docs: publish AI tooling guide and migrate development docs (#2141)
asyncapi/generator: apps/generator/docs/release-process.md

---
title: "Community - Release process"
weight: 230
---

> How releases are produced for the generator packages. For local setup see [Development setup](/docs/tools/generator/development-setup).

Reviewer: ```suggestion
This guide details how releases are handled for the generator packages. For local setup see the [Development setup](/docs/tools/generator/development-setup) guide.
```

To release a major/minor/patch:

## Conventional Commits

For a detailed explanation of conventional commits, refer to [this guide](/docs/tools/generator/contributing#conventional-commits).
To maintain a clear git history of commits and easily identify what each commit changed and whether it triggered a release, we use conventional commits. The feat and fix prefixes are particularly important as they are needed to trigger changesets. Using these prefixes ensures that the changes are correctly categorized and the versioning system functions as expected.

For Example:
```
feat: add new feature
```

## Pull Request Title Guidelines

> To ensure successful workflow execution, all PR titles must follow the Conventional Commits format and PR titles should start with a lowercase character. Incorrect PR titles can cause workflow failures, preventing PRs from being merged.

Reviewer: ```suggestion
To ensure successful workflow execution, all PR titles must follow the conventional commits format and PR titles should start with a lowercase character. Incorrect PR titles can cause workflow failures, preventing PRs from being merged.
```

> For the PR titles you can refer to [this guide](/docs/tools/generator/contributing#conventional-commits)

Reviewer: ```suggestion
For the PR titles you can refer to [this guide](/docs/tools/generator/contributing#conventional-commits).
```
